modules/carousel/Carousel.tsx

```diff
@@ -0,0 +1,275 @@
+  }
+};
+
+const onMouseUp = (e: MouseEvent, scrollElement: any) => {
+  if (dragVariables.drag) {
+    let start = 1;
+    const animate: any = () => {
+      const step = Math.sin(start);
+      if (step <= 0) {
+        window.cancelAnimationFrame(animate);
+      } else {
+        // eslint-disable-next-line no-param-reassign
+        scrollElement.scrollLeft += dragVariables.diffx * step;
+        start -= 0.02;
+        window.requestAnimationFrame(animate);
+      }
+    };
+    dragVariables.drag = false;
+    animate();
+  }
+};
+
+const Carousel: React.FC<ICarouselProps & WithStylesProps<Theme, ICarouselStyles>> = ({
+  cx,
+  styles,
+  items,
+  children,
+  header,
+  footer,
+  scrollArrows = true,
+  scrollableElementRef,
+  marginBetweenItems = 0,
+  useVirtualization = true,
+  rightIcon = "chevron-right",
+  leftIcon = "chevron-left",
+  borderWidth
+}) => {
+  const [elem, setElem] = useState<React.RefObject<HTMLDivElement> | undefined>();
+  const [testCurrentScroll, setTestCurrentScroll] = useState(0);
+
+  useEffect(() => {
+    if (!elem && scrollableElementRef && scrollableElementRef.current) {
+      setElem(scrollableElementRef);
+      scrollableElementRef.current.addEventListener("scroll", () =>
+        setTestCurrentScroll(scrollableElementRef.current?.scrollLeft || 0)
+      );
+    }
+  }, [elem, scrollableElementRef]);
+
+  let newItems = useMemo(() => items.map(children), [children, items]);
+
+  useEffect(() => {
+    if (!scrollableElementRef) {
+      return;
+    }
+
+    addEvent("mousedown", scrollableElementRef?.current, (e: MouseEvent) => {
+      dragVariables.drag = true;
+      dragVariables.diffx = 0;
+      dragVariables.startx = e.clientX + (scrollableElementRef?.current?.scrollLeft || 0);
+    });
+
+    addEvent("mousemove", scrollableElementRef?.current, (e: MouseEvent) => {
+      if (dragVariables.drag) {
+        dragVariables.diffx =
+          dragVariables.startx - (e.clientX + (scrollableElementRef?.current?.scrollLeft || 0));
+        if (scrollableElementRef?.current) {
+          // eslint-disable-next-line no-param-reassign
+          scrollableElementRef.current.scrollLeft += dragVariables.diffx;
+        }
+      }
+    });
+
+    addEvent("mouseup", scrollableElementRef?.current, (e: MouseEvent) =>
+      onMouseUp(e, scrollableElementRef?.current)
+    );
+
+    addEvent("mouseleave", scrollableElementRef?.current, (e: MouseEvent) =>
+      onMouseUp(e, scrollableElementRef?.current)
+    );
+  }, [scrollableElementRef]);
+
+  const scroll = useCallback(
+    (rightDir = true) => {
+      if (scrollableElementRef && scrollableElementRef.current) {
+        const element = scrollableElementRef.current;
+        const currentScroll = element.scrollLeft ?? 0;
+        const scrollElementWidth = element.clientWidth || 0;
+        const halfMarginBetweenItems = marginBetweenItems / 2;
+
+        const sizeOfCardWithMargin =
+          (element.querySelector("*")?.clientWidth || 0) +
```

Comment:
can't you go with something like ".children" or similar ? seems a bit weird to query selector all children with * selector

Reply:
if this is in fact to select the first child